pallets/logion_loc/src/lib.rs

```diff
@@ -200,13 +224,38 @@ pub mod pallet {
 				};
 				<LocMap<T>>::insert(loc_id, loc);
```

Comment:
I would rather do the insert at the end of the function, just before emitting the event. If you keep it like this, you have to ensure in your unit test that this line is correctly rollbacked.

Reply:
Indeed, done.